### PR TITLE
[Config] Remove deprecated macros

### PR DESCRIPTION
**Pull request by alxbilger**
and also removed the attributes associated to the deprecated macros

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/Core/src/sofa/core/objectmodel/BaseData.h [hunk 1]
```diff
@@ -242,25 +230,18 @@ class SOFA_CORE_API BaseData : public DDGNode
     /// @{
     /// True if the value has been modified
     /// If this data is linked, the value of this data will be considered as modified
-    /// (even if the parent's value has not been modified)s
-    SOFA_ATTRIBUTE_DEPRECATED__ASPECT_EXECPARAMS()
-    bool isSet(const core::ExecParams*) const { return isSet(); }
+    /// (even if the parent's value has not been modified)
+    [[nodiscard]]
```
Comment (fredroy) [inline Sofa/framework/Core/src/sofa/core/objectmodel/BaseData.h:233]: Just to highlight it is not "only" removing deprecated macro

File: Sofa/framework/Type/src/sofa/type/Quat.h [hunk 2]
```diff
@@ -117,8 +117,8 @@ class Quat
     /// Convert the quaternion into an orientation matrix
     void toMatrix(Mat3x3 &m) const;
 
-    SOFA_ATTRIBUTE_DEPRECATED__QUAT_API("Function toMatrix(mat4x4) will be removed. Use toHomogeneousMatrix() instead")
-    void toMatrix(Mat4x4 &m) const { toHomogeneousMatrix(m); }
+    SOFA_ATTRIBUTE_DISABLED__QUAT_API("Function toMatrix(mat4x4) has been removed. Use toHomogeneousMatrix() instead")
+    void toMatrix(Mat4x4 &m) const = delete;
```
Comment (fredroy) [inline Sofa/framework/Type/src/sofa/type/Quat.h:121]: ditto

**Review by fredroy — APPROVED** [review]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/framework/Core/src/sofa/core/objectmodel/BaseData.h (hunk 1 above)
Comment (alxbilger) [reply Sofa/framework/Core/src/sofa/core/objectmodel/BaseData.h:233]: you're right ;)

**Comment by damienmarchal** [comment]
In term of deprecation process. I think the workflow was to switch from XXX_DEPRECATED_FEATURE  to a XXX_DISABLED_FEATURE with the function deleted with (=delete). The underlying idea was to keep a track from the "my code is compiling" to "my code is not compiling" but I can go to some declaration to get an idea about what happens.

**Comment by alxbilger** [comment]
You described the usual workflow, but in this case it was not respected. The features should have been disabled in v22.06.

**Comment by damienmarchal** [comment]
> You described the usual workflow, but in this case it was not respected. The features should have been disabled in v22.06.
Before the release we discussed during a sofadevmeeting that it was not wise to disable the features at release time. The final agreement was on the "timing" of action (before or after the release) and not about the conformance to the three step workflow (deprecate-disable-remove) we have. 

A lot of people don't read warning(deprecation) messages and only fix their code base when it does not compile I think we should stick to the workflow.

**Comment by alxbilger** [comment]
To do: Revert changes from BaseData (because addressed in another PR)

**Comment by alxbilger** [comment]
Replaced by https://github.com/sofa-framework/sofa/pull/3346 and https://github.com/sofa-framework/sofa/pull/3279

**Pull request closed without merging**